lib/plottr_components/src/components/timeline/CardCell.js

```diff
@@ -17,7 +18,89 @@ const CardCellConnector = (connector) => {
   const CardAdd = UnconnectedCardAdd(connector)
   const ErrorBoundary = UnconnectedErrorBoundary(connector)
 
-  class CardCell extends PureComponent {
+  class CardCell extends Component {
+    constructor(props) {
+      super(props)
+      this.state = {
+        dragging: false,
+        inDropZone: false,
+        dropDepth: 0,
+        hovering: false, // forces a rerender to determine where to place the overlay
+      }
+    }
+
+    handleDragStart = (e) => {
+      this.setState({ dragging: true })
+      e.dataTransfer.effectAllowed = 'move'
+      const cardIds = this.props.cards.map((card) => card.id)
+
+      if (!Array.isArray(cardIds) && cardIds.length <= 1) return
+      e.dataTransfer.setData('text/json', JSON.stringify({ cardIds }))
+    }
+
+    handleDragEnd = () => {
+      this.setState({ dragging: false })
+    }
+
+    handleDragEnter = (e) => {
+      // https://www.smashingmagazine.com/2020/02/html-drag-drop-api-react/
+      if (!this.state.dragging) this.setState({ dropDepth: this.state.dropDepth + 1 })
+    }
+
+    handleDragOver = (e) => {
+      e.preventDefault()
+      if (!this.state.dragging) this.setState({ inDropZone: true })
+    }
+
+    handleDragLeave = (e) => {
+      if (!this.state.dragging) {
+        let dropDepth = this.state.dropDepth
+        --dropDepth
+        this.setState({ dropDepth: dropDepth })
+
+        if (dropDepth > 0) return
+        this.setState({ inDropZone: false })
+      }
+    }
+
+    handleDrop = (e, allowDrop, cardIndex) => {
+      e.stopPropagation()
+      e.preventDefault()
+
+      if (this.state.dragging) return
+      if (!allowDrop) return
+      this.setState({ inDropZone: false, dropDepth: 0 })
+
+      const json = e.dataTransfer.getData('text/json')
+      const droppedData = JSON.parse(json)
+      if (!droppedData.cardId) return
```

Comment:
What if it's a stack of cards?
Wouldn't `droppedData.cardIds` be bound then?

Reply:
Did you drop a stack of cards onto another stack of cards in your testing?

Reply:
Yeah i have tried but its not allowed yet. Im sure we i should also allow stacked cards to be droppable to another stack of cards 😅 wdyt?

Reply:
I think it should be allowed.  It seems unexpected to forbid that but allow one card to be dropped onto a stack. :)